meshmode/array_context.py

```diff
@@ -368,6 +387,41 @@ def tag_axis(self, iaxis, tags: Union[Sequence[Tag], Tag], array):
         .. versionadded:: 2021.2
         """
 
+    @memoize_method
+    def _get_einsum_prg(self, spec, arg_names, tagged):
+        return lp.make_einsum(
+            spec,
+            arg_names,
+            options=_DEFAULT_LOOPY_OPTIONS,
+            tags=tagged,
+        )
+
+    def einsum(self, spec, *args, arg_names=None, tagged=()):
+        """Executes a :mod:`loopy` einsum program on the arguments *args*.
```

Comment:
```suggestion
        """Computes the result of Einstein summation following the
        convention in :func:`numpy.einsum`.
```
(The important bit here is to specify semantics, not implementation details. The docs have no business stating *how* this is to be done ("via a loopy program"? "via a loopy program"... this will be false for the lazy array context.)